### DISCUSSION TITLE
A Major Proposal Between Projects

### DISCUSSION THREAD

**Discussion by generic-pers0n** [tenacityteam/tenacity-legacy#733] [General]
**TL;DR**: Maybe we can merge parts of Tenacity into Saucedacity or the two together under a single audio editor. However, Saucedacity is moving to Qt requiring quite a large rewrite of code and we require GPLv2 or later for contributions, so please be aware of that if you guys want to move forward with this.

---

*Please read this post carefully as it is a little long...*

So it came up in Saucedacity that one is suggesting to merge Tenacity's translations into Saucedacity here. That's an idea, and hopefully we can get on to it at one point. However, another idea had spawned in my head, and so I want to take a look at this...

What if we merge Tenacity into Saucedacity? Or what if we merge the two into one?

I've actually been thinking of this, but I haven't been confident about proposing it. Now that I am, I want to offer this because I see that there have been some good contributions to Tenacity by some good people. I also value the work of these contributions too, so if you guys want to continue on your magnificent contributions, here's an opportunity for you.

Of course, this is up to you guys. I might already merge translations given an issue regarding translations over at Saucedacity, but for anything else, it is entirely up to you all if you want to proceed with this.

Before we get too far, I'd also like to mention that most of the things in Saucedacity are going to be subject to a rewrite in order to move to Qt, as is a part of our project vision. Saucedacity:s Qt transition should be complete by a 2.0 release of Saucedacity, but we will be rewriting quite a couple of things. Additionally, for contributions to Saucedacity, we currently require your contributions are under GNU GPL v2 or later (as this has always been the case since the start). I would like to bring this up because while we may have the same spirits, we might have different project directions. I would like you guys to consider this first so you don't regret anything later on. If you have questions, please ask them (even if you think they're stupid ones, it is definitely OK to ask them all).

When the Audacity controversy first unfolded a year ago (at this point), there spawned a total of 4 forks (Saucedacity and Tenacity included). We all had the same spirit, but there was no clear alternative. Back then, unity was needed, and still to this day. I recognize that this proposal comes over a year late, but better late then never I guess. I also recognize that I might never be able to unite everyone around the same campfire, and we may never achieve full unity, but I hope that we can at least start to work together if not merge in any way.

**Comment by Be-ing** [comment]
The code is here in the repo. You're welcome to cherry pick/merge it if you want, though I think it'd be up to you to do the work.

**Reply by generic-pers0n** [reply]
For sure! I'm already going to be looking at translations for now. Anything else I'll have to look at first.

As for me having to do the work, I'll be up for doing any work should I choose to take anything from Tenacity. For now, I'll take a look at translations.

**Comment by leio** [comment]
My take on things can be read from https://github.com/tenacityteam/tenacity/issues/572#issuecomment-916423566

**Comment by n0toose** [comment]
I've been waiting for this and was disappointed this did not happen from the beginning.

Well. We're dead, so, you have my absolute endorsement to use whatever patch I created as long as long as you preserve the DCO notes and, if a commit is to be modified, `Co-authored-by`.  I believe you can reuse most of our work on build systems, packaging, contribution guidelines etc. -- I also believe that we tried to organize ourselves with the edge case scenario of us "dying out" and another person taking our work and moving further with it in mind.

As far as merging things *into* Tenacity and/or using our organization: Even if we have the 7.2k star "clout" (even if this place is being persistently spammed by total weirdos), I am not sure if I can give you access to the repositories (or even admin access) as @emabrey and @Semisol are also administrators of this organization and also have to agree. I would personally be glad to hand over the reigns/admin access to people that remain active on here (such as @Be-ing, who manages other repositories under this organization who would probably appreciate the extra control without way too many people that have nothing to do with the project anymore causing anxiety).

If you do not want to use our organization, I would recommend you to get out of GitHub while you can, as it made maintenance a living hell for me and others when we used it.

---

Another issue I would like to bring up is that, in the afterlife, we have some ghosts to deal with: We have accumulated a lot of money under the 501(c)(3)-controlled [Open Collective](https://opencollective.com/tenacity), with people (still!) donating despite our inactivity. The money cannot be abused, as no expenses can be made without Open Collective's explicit approval, as they are the ones responsible for the paperwork and also the ones to face consequences if that were to happen, which was one of the reasons why we went for it at first. In retrospect, I would not do that again. Our only attempted expense was for paying for SourceHut-related services, which TL;DR was not possible because of @OpenCollective's then-allowed ways of issuing expenses and because of SourceHut's payments methods, as I was told.

That account is solely controlled by @emabrey. When I reluctantly agreed to have such a thing (which kind of hurt us, because we never did that and scared away contributors full of knowledge because we had not yet established a well-tested governance structure and because with money come expectations, I quit being against it that under the condition that if our project were to die and/or go to hiatus, we would donate the money to another Linux audio-related project. This part of the deal was mentioned on [our GitHub Sponsors page](https://github.com/sponsors/tenacityteam), but I am not able to do anything to carry it out right now. I will talk to @opencollective about this in a few weeks if this issue does not get resolved.

I think I clarified all of my thoughts on this subject expecting responses, I may not respond in time because I am currently in the middle of an exam phase.

(As essays tend to attract attention, I will remind any readers that **this place here is a place of work, not an imageboard or Reddit**, and disruptive comments will be removed by someone with permissions to do so.)

**Reply by n0toose** [reply]
Sorry for writing this block of text, it's just that I've had enough of that and I am trying to be very careful about what I say in public in respect to this project.

**Reply by generic-pers0n** [reply]
@n0toose I definitely see what you are saying.

Now, I am currently not interested in any admin (or at the very least write) access. I am also not interested in taking donations either, neither would I want to receive any existing donations. Where these donations would go, however, I don't know myself either. This should be talked over with @opencollective as you've suggested.

There is also an existing organization, @saucedacity, which I am currently admin of (as I created that organization). Anyone can join if they are interested on working on Saucedacity or similar (e.g. downstream audio libs for Saucedacity, a website, etc). As for your organization, again, I'm not interested for now. **But, if some think it's a good idea for me to be a part of it, we should discuss that first**. ~~Actually, I might have more to say, so I'll likely make another post.~~

If, at this point Tenacity is dead, we might be talking about merging most of it in Saucedacity at this point. I am already eyeing a couple of things to merge into Saucedacity. As for the DCO notes, those will be preserved. (Actually, I might start to require them out of your guidelines). In Saucedacity's future stable release (i.e., 1.3 or later), I will be eyeing what to merge into. Maybe not as of 1.2 because that's in beta, and everything is close to being finished for a stable release.

Overall, I'm interested in carrying on the work that has started here in Tenacity, but we cannot avoid any ghosts or surrounding issues that currently remain.

Finally, good luck on your exam. And don't worry about responding in time. You're good :)

**Reply by n0toose** [reply]
> Overall, I'm interested in carrying on the work that has started here in Tenacity

You're free to take it, if you need any direction or to ask someone who has attempted the exact same thing, you can definitely message me on Matrix as I'd love to help out in whichever capacity I can.

**Reply by n0toose** [reply]
I think I speak for all of us, because a) that's the part of the deal that we gave people when we started writing the code under our license, and b) because your project is a continuation of values that are similar to ours, and seeing some of that pretty awesome work that other people did on this repository (such as the themes) would be pretty nice, especially if it's by a craftsman that manages to deal with an entire beast of a codebase on their own.

I think we would've done better, if we did not have all of this attention we also have had to deal with from the get-go before producing a working product. But yeah, dunno, just let me know if you need anything.

**Reply by emabrey** [reply]
I haven't dealt with the OpenCollective issue yet because I've been busy trying to switch into a new career path. That said, all the money is still safely within the account. The current balance as of writing this comment is $559.31 USD.

**Comment by generic-pers0n** [comment]
So based on the situation, I **initially** decided to go ahead and merge some things into Saucedacity.

However, our name came up in [this issue](https://github.com/saucedacity/saucedacity/issues/21), and we've been discussing it. @TheEvilSkeleton had possibly suggested that we carry over the name.

Overall, I would like to see what you guys think about a name transfer. I don't know if this would be a good idea, and this is clearly up to you guys. I believe that a name change from "Saucedacity" is a good idea at this point, and I might open an issue in Saucedacity's issue tracker for name suggestions. I believe that a separate name might be a good idea, and I might be veering towards that.

**Reply by n0toose** [reply]
I was about to say that this is OK, but then I started authoring this announcement:

<details>
<summary>Original message I began authoring, not applicable</summary>
I am OK with it, but I have not heard from any other admins and contributors and do not have explicit permission to use my control to e.g. make people admins.

Considering that this is a public forum and notifications there, provided that @generic-pers0n wants to go for Tenacity, I will give it **one week** for other admins to respond. If I do not see any response, I will get in touch with Open Collective about "recovering" the funds and starting a discussion as to how they will be used with previous contributors and will do things such as manage the repositories and provide @generic-pers0n with elevated permissions in this organization.

My full control and responsibility extends to the GitHub organization and the IRC channels. The problem is that using Tenacity implies that our own organizations and infrastructure should be used.
</details>

I think the can of worms here is a bit much, I can definitely handle all of that for you if you want me to without any effort from your part.

**Reply by Be-ing** [reply]
I'd only be okay with transferring the name if you also take (most) of the code from Tenacity.

**Reply by n0toose** [reply]
I presume this will turn out to be the case, regardless of whether the name is chosen or not.

**Reply by Be-ing** [reply]
I am not presuming anything until it happens.

**Reply by generic-pers0n** [reply]
> I am not presuming anything until it happens.

This should be happening soon. I just released 1.2 and will go on to merge bits of Tenacity into the Saucedacity code base.

**Edit**: More details:

* Once I get things merged into Saucedacity's code base, I will start to eye on changing Saucedacity's name to Tenacity. This is only after things are merged, so that should happne later.

* Along the time, I will look to merge some open and draft PRs into Saucedacity's code base also. Open PRs we can discuss, and draft PRs we might be able to finish.

**Comment by abrolag** [comment]
I've been out of the loop for a long time, and was saddened to see what's happened. However I would also vote for taking the name across as well as the code. It's more memorable and would probably appeal more to less techie people.

**Comment by generic-pers0n** [comment]
**UPDATE 2022-08-02: I have made some final decisions and some more proposals. I have edited my post accordingly. Please (re-)read carefully.**

---

Alright, so I am very close to making final decisions regarding this entire situation. It looks like people are supporting the carrying over the name "Tenacity", so I will now use your guys' name given the support.

Additionally, after taking advice from @n0toose, I have made the following plan that will be enacted very soon, given feedback:

1. I will make one last release under our old name "Saucedacity". This will be our awaited 1.2 release. I want to make this release under our old name because I don't have plans to merge anything from Tenacity in 1.2 since I have been waiting to get 1.2 out for some time. Therefore, there's really no work from Tenacity and would not want to use the name "Tenacity" yet. Soon after, however, I will start to merge parts of Tenacity into the project and we will carry the name over. Then, Tenacity should see it's first release at 1.3 or something (there will be some history behind the version number)

2. I have heard some suggestions on moving away from GitHub. Once the project gets large enough, we will start the process of moving and I will make an announcement. In the mean time, I will make another announcement saying that we will have plans to move to another code hosting platform in the future after our 1.2 release.

Things from Tenacity that I am currently eyeing:

* Translations
* Build system
* Themes

If you guys have any input, let me know. I hope to finally begin this long awaited suggestion very soon, given how late this suggestion was made in the first place.

---

**Note about admin access**: I haven't decided if we should continue on development in this repository or in Saucedacity's repository. Things would have to be renamed of course, but I propose my repo because the Saucedacity codebase is already there. This repository can be renamed appropriately.

In terms of organizations, we can transfer Saucedacity's repository to Tenacity's organization (@tenacityteam) if that's easier. Then, we can start merging changes from there. If we decide to use @saucedacity instead, however, I would rename the organization to "Tenacity" or something (to not conflict with "Tenacity Team". Additionally, if there are any admins that want to be active in the new organization, I would be glad to re-add you. Just contact me either through email or through Matrix.

**Comment by poyribn** [comment]
i have seen Codeberg suggested as the future Git platform for Saucedacity.
this seems like a good choice to me, since they're dedicated to supporting only projects with the open-source spirit, and ban proprietary projects from Codeberg.

so putting Saucedacity on Codeberg would reassure contributors — new contributors especially — that Saucedacity is a true open-source project instead of a limited rights thing like Unreal Engine.

also i am biased since i put my own Git projects on Codeberg.

**Reply by Be-ing** [reply]
Codeberg does not provide Windows or macOS CI. If you decide to stop making Windows and/or macOS builds, go for it. Otherwise be prepared to spend a bunch of time setting up your own cross platform CI instead of writing code.

**Reply by generic-pers0n** [reply]
> Codeberg does not provide Windows or macOS CI. If you decide to stop making Windows and/or macOS builds, go for it. Otherwise be prepared to spend a bunch of time setting up your own cross platform CI instead of writing code.

Thanks for the info. I have heard about this and I am okay with dropping macOS builds. Windows, however, I can always test locally in a VM I have set up, so if (or when, at this point) we move to Codeberg, I will be testing more often in my Windows VM.

**Comment by generic-pers0n** [comment]
Although I'm still awaiting feedback, I have done probably the last commit to Saucedacity 1.2. I am expecting Saucedacity 1.2 to be released tomorrow (August 3, 2022).

**Reply by generic-pers0n** [reply]
More about the situation:

After that release, I will start to merge things into Saucedacity's code base. I am eager to build upon the work you guys have started, so I'll do that as soon as I can!

In terms of name changes, I will proceed with preserving the Tenacity name and will rename Saucedacity accordingly. However, I would like to make it clear which organization we are going to use first before we make that change. I believe we should move my repo under this organization, rename it, and rename this existing repo. However, I can always rename my organization to just "Tenacity" and also rename the Saucedacity repo accordingly. I will also add anyone that was an admin in @tenacityteam to my organization should we choose this route. Also, to clarify, given the decision to move to Codeberg or so in the future, this is only contemporary. This decision is up for you guys to make, and I will be happy with either one :). Afterwards, I will start rebranding efforts.

After all of that, our main focus will be to make a new release featuring all of the work from Tenacity. I haven't decided if this should be 1.0 or 1.3 (given that this will be Saucedacity's code base), but an audio editor under the name "Tenacity" deserves it's first release in all of time, especially after a year in :)

Again, if you have feedback, let me know. If anyone wants to discuss things over Matrix, feel free to reach out to me. I'll answer any questions and follow any requests the best I can from you guys.

**Reply by emabrey** [reply]
> Again, if you have feedback, let me know. If anyone wants to discuss things over Matrix, feel free to reach out to me. I'll answer any questions and follow any requests the best I can from you guys.

What is your matrix username?

**Reply by generic-pers0n** [reply]
> What is your matrix username?

My Matrix username is `@generic-person:matrix.org`

**Reply by abrolag** [reply]
Pardon my ignorance but what *exactly* is the issue with github - apart from the general unease with Microsoft control?
Also have you considered sourceforge. It's far more stable (and friendly) than it was a few years ago.
As an aside the project I work with is hosted on both sourceforge and github - independently so neither is reliant in any way on the other.

**Reply by generic-pers0n** [reply]
> Pardon my ignorance but what _exactly_ is the issue with github - apart from the general unease with Microsoft control? Also have you considered sourceforge. It's far more stable (and friendly) than it was a few years ago. As an aside the project I work with is hosted on both sourceforge and github - independently so neither is reliant in any way on the other.

I've seen various reasons as to why GitHub isn't the best platform and its shortcomings. One example includes moderation, and another is commit history (which I have unfortunately saw the hard). There are many reasons to go on, and I'm sure others would like to give their share on it too.

About SourceForge, I have mixed feelings on SourceForge. On it's page of open source projects, there are some projects that haven't been updated within a couple of years (e.g. 2019, 2016, even 2014) to those who have just been updated within a couple of days. Personally, whenever I visit a link to a project on SourceForge, it's always been a project that hasn't been updated within years. However, modern SourceForge is a possible option.

Also, to inquire, how does this projected hosted across GitHub and SourceForge? We might be able to do this with whatever alternative hosting platform is finally decided. Again, this looks to be possibly Codeberg.

**Comment by akleja** [comment]
Hi!
I've been gone for a long time, but this seemed like the time to pop back in.

I think this whole endeavor seems hopeful. Merging these projects makes sense to me.

For me what made things difficult with this project was the fact that we decided to rely on audacitys library separation efforts. Merging the patches was not an enjoyable effort. And it made any development on our part difficult. After a while it felt kind of meaningless to make changes or additions without starting to separate things out. Most of the solutions seemed like temporary quick fixes to have something presentable. Plus each change ensured future merging of patches would get even more time consuming. And as time went by the idea of catching up with the merging seemed more and more distant. Plus everything seemed a bit hectic with all the attention. I also had some health concerns that forced me to shift focus away for quite a while, and when I tried to get back into the project it just seemed overwhelming.

But in the meantime I've gained some more programming experience since I was last active in this project, but my time is also a bit more limited these days. With all that said I'd be glad to try to contribute a bit again if my efforts would be needed. But I probably won't be of much help with the whole merging process. I'll gladly answer any questions that might come up though. I'm mostly familiar with the parts of the code relating to themes and some UI stuff, but have a general grasp of some of the other parts as well.

**Reply by generic-pers0n** [reply]
**TL;DR**: I don't think there will be much for you to do during the merging process (though I will reach out to you if I have questions) but you are definitely going to be needed for upcoming tasks. You are also free to open or draft new PRs during the merge process.

---

Hello! First off, welcome back! I hope everything is going well with you given what you've said, so let's hope everything's okay there. Your health is very important, even more important than any open source project arguably.

As I've been browsing around this repo, I've come across some of your work and I'm quite amazed by it! Your work is very valuable and is also something I want to carry forward too. But first, there is something that you said that I have some thoughts on:

> For me what made things difficult with this project was the fact that we decided to rely on audacitys library separation efforts. Merging the patches was not an enjoyable effort. And it made any development on our part difficult. After a while it felt kind of meaningless to make changes or additions without starting to separate things out. Most of the solutions seemed like temporary quick fixes to have something presentable. Plus each change ensured future merging of patches would get even more time consuming. And as time went by the idea of catching up with the merging seemed more and more distant. Plus everything seemed a bit hectic with all the attention. I also had some health concerns that forced me to shift focus away for quite a while, and when I tried to get back into the project it just seemed overwhelming.

I totally see this, given one time that I wanted to merge Audacity 3.0.5's changes in Saucedacity at one point. Ultimately, I gave up due to linking issues with FFmpeg, and I became more conservative when it comes to those changes. More recently, I merged `lib-ffmpeg-support` (but into `src/` instead, and that required some work to be done, but I got it done. However, it was initially as a test, and when it fixed an issue with FFmpeg, I kept it. Additionally, I now realize that the initial burst of attention you guys got hurt the project in the long run because of the pressure to have something presentable. Unfortuantely, this never happened. I have a set vision for a new release under Tenacity's name featuring parts of Tenacity and Saucedacity. This vision will be put into a release agenda, which I will release **unfinalized** so we can decide what we want for our next stable release.

Finally, as you've said, I believe that there won't be much for the merge process, so you might not bee needed while we merge things into Saucedacity's code base. However, after things are merged, we will be glad to have you back contributing to the project! I already have a couple of things planned, both during and after the merges. One of them is the planned migration to Qt, which will start after we merge the build system. This is intended to go alongside development of Tenacity 1.x, so the process towards 2.0 will be slow. However, if you want to start writing code for a new code base, go ahead! The only thing requirement that I plan on imposing is requiring GPL v2 or later code for the new code base of Tenacity 2.0. You can start contributing to this code base once we get the new build system merged if you want, or you can contribute to our existing one. During the merging process, we'll also be accepting new PRs too, so if you want to open or draft a new PR, whether it's an existing or new one, go ahead! Either way, however you want to contribute is up to you. Regardless, you are a valuable contributor to have with the project and we will definitely need your help with things. We hope to be working with you (and other contributors) again towards the first stable release under the name "Tenacity"!

**Reply by akleja** [reply]
Thanks!
No worries, I'm doing fine, but thanks for your concern!

I guess I've been fiddling around a bit here and there with a bit of this and that :)

The merges sure was a bit of a headache in more ways than one. I haven't kept up at all but assume that is code for compatibility with ffmpeg 5?
It sure was a bit much. Okay yeah sounds like a good plan to me.

I'll keep out of the discussions relating to code forges, and other administrative stuff since I don't feel I have too much to add there.
But I'll start by checking out Saucedacity during the weekend, right now I've only glanced at it. Just don't expect too much from me at first.
I will probably try to help with 1.x, but the 2.0 migration project seems extra interesting I must say.
I hope this is going to be as fun as it seems!

**Reply by generic-pers0n** [reply]
Unfortunately, the code wasn't for FFmpeg 5 support in Saucedacity. We will have to merge a newer version of `lib-ffmpeg-support` in the code base. That might happen later.

Now, to add, I plan to move to using Tenacity's name, but we'll likely have our next release as 1.3. Until the rename happens, I will refer to Saucedacity as, well, Saucedacity.

Finally, I also hope this project will be as exciting as it seems too! I'm quite excited myself also, and I can't wait to see what's in store.

**Reply by akleja** [reply]
Okay!

Yeah that makes sense.

I took a quick glance at the commit history for Saucedacity, and I must say it makes me hopeful.

**Comment by generic-pers0n** [comment]
More things now that 1.2 has been released:

* The next release I have planned using the Tenacity name is Tenacity 1.3. We will merge as much things as we possibly can, but the main priority is bug fixes and focusing on a stable release. Maybe during this time we can merge some previously unmerged PRs, but that is variable. They would be guaranteed for a future release, however.

* A release agenda will be released soon after to detail what is planned for the new release. I will request feedback for this release agenda once it has been posted.

Plus, there are some things about organizations I would like to express:

* I think we are likely going to use the @saucedacity organization. In that case, we will rename that accordingly to "tenacity"). If you guys want to use a different organization or maybe @tenacityteam instead, let me know and we can arrange things.

* If any current Tenacity admin wants to become an active admin for my organization, open a new discussion or issue in the Saucedacity repo and I'll get right to it. The same applies to any interested contributors that have been active with Tenacity and want to be active with the new project.

**Reply by n0toose** [reply]
"Tenacity" is taken and GitHub does not release dormant usernames anymore, not sure how you want to go along with this.

Keep in mind that if you move a repository, GitHub will redirect people using the old link to the new link until a new repository takes the place of the old link. I'd be pretty OK with giving all relevant access to @tenacityteam and now that I have re-established a means of contact with the other admins as well, things should be pretty OK.

**Reply by caughtquick** [reply]
Wouldn't the decision to allow @generic-pers0n to continue the development of Tenacity in its current state with an effort to merge changes from Sauceacity be a better approach. It would also allow Tenacity to have a future and give a much better solution for the funds in the OpenCollective.

**Reply by generic-pers0n** [reply]
> "Tenacity" is taken and GitHub does not release dormant usernames anymore, not sure how you want to go along with this.
> 
> Keep in mind that if you move a repository, GitHub will redirect people using the old link to the new link until a new repository takes the place of the old link. I'd be pretty OK with giving all relevant access to @tenacityteam and now that I have re-established a means of contact with the other admins as well, things should be pretty OK.

I now believe that giving me admin access is perhaps the best option here.

**Reply by caughtquick** [reply]
I agree but that's a decision up to @Be-ing @Semisol @emabrey @n0toose

**Reply by generic-pers0n** [reply]
> I agree but that's a decision up to @Be-ing @Semisol @emabrey @n0toose 

I do see that this is their decision. Therefore, it is up to them to make that decision.

**Comment by abrolag** [comment]
On Thu, 04 Aug 2022 17:31:18 -0700
Avery King ***@***.***> wrote:

>Also, to inquire, how does this projected hosted across GitHub and SourceForge? We might be able to do this with whatever alternative hosting platform is finally decided. Again, this looks to be possibly Codeberg.
>
By virtue of being the one with the most free time (retired) but ironically the
least experience, I'm nominally the lead developer.

As well as adding code myself, which I then push to both github and sourceforge
via separate ssh connections, if a pull request comes from github, after
verifying, accepting and updating my copy I then push it to sourceforge.
Similarly accepting from sourceforge then going to github.

If either repository is lost or becomes corrupted there is a secure copy still
available as there is no direct connection between them. If there was a total
meltdown with my setup, there are two other trusted devs with admin rights -
this has not been needed in nearly 10 years.

P.S.
Obviously, the viability of this method depends on the size and complexity of
the project!

-- 
Will J Godfrey {apparently now an 'elderly'}
https://willgodfrey.bandcamp.com/
http://yoshimi.github.io
Say you have a poem and I have a tune.
Exchange them and we can both have a poem, a tune, and a song.

**Reply by n0toose** [reply]
We tried that with SourceHut. Mirroring things on two places did not work, as e.g. merging pull requests from GitHub adds, for example, a number corresponding to the issue/pull request. This only works on GitHub.

SourceForge is not a popular option anymore for many reasons, and network effects matter in our case.

**Comment by generic-pers0n** [comment]
Alright, so an update on how the merges are going...

Merging things from Tenacity into Saucedacity has required me to take a look at Tenacity's code base. I now know a little more of Tenacity's code base, and I have encountered a few issues. Most recently, I have been working on merging Tenacity's build system. Using this example, I have realized that merging things may require some effort.

In the case of Tenacity's build system, I have discovered the `build_rewrite` branch. I have been attempting to trace commits that lead up to the commits introduced by this branch and have gotten some progress (I have already cherry-picked one so far, at the very least). However, I have realized that there is one main difference: Saucedacity doesn't directly depend on libsoxr, so we don't include it as a dependency. I believe this can easily be fixed. However, the main problem was getting wxWidgets to work. Although I had wxWidgets 3.2 from CodeLite installed on my system, I was not able to get wxWidgets to work. I am currently investigating this.

Overall, I've realized that there will be some differences between our code bases where more work might be required in order to completely merge them without conflict, so I might be asking some questions. For now, I'll see what I can do.

---

**EDIT**: I just realized the commits I'm cherry picking don't quite work on my system in Tenacity either...

I've checkout some of these commits from the `build_rewrite` branch that was merged into Tenacity and Tenacity doesn't build. This is regardless if vcpkg is being used or not. So it appears that I have a way to go before we get this new build system up and running...

**Comment by generic-pers0n** [comment]
Since this repo was unarchived, I'd like to post an update here for anyone still monitoring this discussion.

We have so far merged Tenacity's themes and dynamic compressor, with the build system being worked on and translations to be added soon after rebranding efforts continue. (I wanted to save some work on renaming strings with "Tenacity" in them). The build system merge will take some time, but once it's ready, it'll be merged. However, over new releases, more is to come afterwards.

I have made an announcement here detailing everything that could happen soon if no one objects: https://github.com/saucedacity/saucedacity/discussions/72

This has gone quite a long way, and we're not too far off from a planned Tenacity 1.3, Tenacity's first ever stable release!

**Comment by abrolag** [comment]
Thanks for keeping us informed. Looking forward to trying the fully merged version.
